lib/std.dl

```diff
@@ -77,29 +77,50 @@ extern function str_to_lower(s: string): string
 extern function hash64(x: 'X): bit<64>
 extern function hash128(x: 'X): bit<128>
 
+/* The `Group` type is used exclusively in aggregation operations.  It
+ * represents a non-empty list of objects sorted in ascending order.
+ */ 
+extern type Group<'A>
+
 /*
  * Standard aggregates
  */
 
-extern type Group<'A>
+/* Returns the number of elements in the group, which is guaranteed
+ * to be >0. */
+extern function group_count(g: Group<'A>): bit<64>
+
+/* Returns the first element of the group.
+ * It always exists, as aggregation cannot return an empty group. */
+extern function group_first(g: Group<'A>): 'A
+
+/* Returns `n`th element of the group if the group is large enough
+ * or `None` otherwise.
+ */
+extern function group_nth(g: Group<'A>, n: bit<64>): Option<'A>
 
-extern function count(g: Group<'A>): bit<64>
 extern function group2set(g: Group<'A>): Set<'A>
 extern function group2vec(g: Group<'A>): Vec<'A>
 extern function group2map(g: Group<('K,'V)>): Map<'K,'V>
-extern function group_unzip(g: Group<('X,'Y)>): (Vec<'X>, Vec<'Y>)
 extern function group_sum(g: Group<'A>): 'A
 
+function group_unzip(g: Group<('X,'Y)>): (Vec<'X>, Vec<'Y>) = {
```

Comment:
This would make sense for other datatypes that can be iterated upon. But do you want this to be a primitive? The generalization would take functions from elements to values instead of projecting on components.

Reply:
Unfortunately, we do not support lambdas or even function-typed variables at the moment.  This is only a primitive in that it is declared in the `std` lib; it is not a primitive in that it is actually implemented in DDlog.  Is your point that it is kind of arbitrary and should not be in the standard lib? I think I only use it in one of the projects at the moment and could just move this function there. But on the other hand, I don't see any harm in keeping it in `std`.  In fact, even Haskell, which support lambdas, provides unzip in the standard library.